docs/source/dev/start.md

```diff
@@ -0,0 +1,130 @@
+
+<!--
+SPDX-FileContributor: Michael Meinel
+SPDX-FileContributor: Oliver Bertuch
+-->
+
+# Tutorial: Start to develop HERMES
+
+Following this tutorial will guide you along setting up a local copy of the code for development of the workflow itself.
+This is not necessary if you only want to use HERMES or want to develop an extension plugin loaded at runtime.
+
+```{note}
+In the future, there will be a tutorial how to extend HERMES with a plugin available.
+```
+
+## Prepare your environment
+
+First, install Python 3.10 and activate your Python environment of you choice (e.g., `venv`, `conda`, `pipenv`, ...).
+This is very dependent on which Python environment you prefer.
+
+Next to Python 3.10 you need to [install `poetry`](https://python-poetry.org/docs/#installation) inside your environment
+to manage and install dependencies. Please note we require using `poetry>=1.2`.
+
+If you use `conda`, the above can be achieved with the following commands:
+
+```shell
+conda create --name hermes python=3.10 poetry
+```
+
+You should confirm whether to proceed.
+
+## Get the source code
+
+Next, you need to obtain a version of the HERMES source code.
+
+You can either download it as a zipped package or clone the whole Git repository.
+You can clone the repository and enter the project directory as follows:
+
+```shell
+git clone https://gitlab.com/hermes-hmc/workflow.git
+cd workflow
+```
+
+## Learn how our repo is structured
+
+- All Python sources are in the `src` folder
+- `pytest` tests are in the `test` folder
+- [Architectural Design Records (ADR)](https://adr.github.io/) are in `docs/adr`
+- API documentation is automatically generated into `docs/source/api`
+- All other Sphinx-based documentation lives in `docs/source/*`
+
+This project uses 
+
+- a development branch (`develop`) to merge developments into, this is the default branch
+- actual development is done on "feature" branches (this includes non-feature work such as bug fixing), see also our
+  {doc}`contribute`.
+- a `main` branch which only includes releases
+
+## Install HERMES and dependencies
+
+To finally install all the required dependencies and HERMES itself, you can now use `poetry`:
+
+```
+poetry install
+```
+
+### Which dependencies do we use?
+
+Our feature implementations build upon these main dependencies:
+
+- [`click`](https://click.palletsprojects.com/) for providing a commmand line interface
+- [`jsonschema`](https://python-jsonschema.readthedocs.io) to validate different JSON based file formats
+- [`ruaml.yaml`](https://yaml.readthedocs.io) to parse and validate YAML files
+- [`cffconvert`](https://github.com/citation-file-format/cff-converter-python) to harvest data from the CITATION.cff file
+
+Finally, there are some dependencies that are only required for active development. These are:
+
+- `pytest`, `pytest-cov` to run the developer tests and monitor the test coverage
+- `Sphinx`, `myst-parser` to build the documentation
+- `flake8` to check the coding style
```

Comment:
```suggestion
To learn what other packages `hermes` depends on, have a look at the project configuration file `pyproject.toml` (in the root of the repository), or use `poetry show --only main`. 

To show dependencies that are only required for active development, or for building documentation run `poetry show --only dev` and `poetry show --only docs` respectively.
```

This list is basically unmaintainable. Show people how to use their tooling instead.

Reply:
I somewhat agree. Yet I'd still include mentioning our main dependency of `Click` and `Sphinx`, as this is a) unlikely to change and b) an opinionated choice among the available larger frameworks to create and document a CLI app.

Reply:
Mentioning click: okay.
Mentioning Sphinx: inconsequential, as this is the default way to document Python projects.

Reply:
Alright, fair enough. Will rephrase text local and push again.